Secrets header version badge

zofskeez opened the pull request:
Small change to always show the version badge in the secrets engine header.

![image](https://user-images.githubusercontent.com/24611656/139942697-de983152-dd46-4799-89a0-32ac273361b5.png)

hashishaw commented on ui/app/templates/components/secret-list-header.hbs:
```diff
@@ -16,11 +16,9 @@
       <h1 class="title is-3">
         <Icon @glyph={{or @model.engineType "secrets"}} @size="xl" class="has-text-grey-light" />
         {{@model.id}}
-        {{#if (eq @model.options.version 2)}}
```
Only KV type backends should have this badge. This component is used in every type of backend, so we still need some check here to make sure we're only rendering on mount type KV

zofskeez replied:
Thanks for pointing that out @hashishaw. I'll update and add a test. Quick question regarding `engineType`. I see in some places there is check for both `kv` and `generic` types. Is `generic` a pre kv type from older versions that we are now mapping to kv?

hashishaw replied:
That is a great question. Let me check and get back to you 👀

hashishaw replied:
Yes, `generic` was "renamed" to `kv` but we keep it for backwards compatibility 👍

hashishaw reviewed (changes requested):
Might be worth adding a little test too

zofskeez commented:
@hashishaw I made the change to only show the badge for `kv` and `generic` engine types and added a test. I also added a mirage factory for the `secret-engine` model as well as a test helper to return ember data models when given models generated by mirage. This could be convenient in integration tests when the component expects a model as an argument rather than stubbing out a pojo.

hashishaw reviewed (approved):
Great work, thank you!
